web/packages/design/src/SlideTabs/SlideTabs.tsx

```diff
@@ -16,40 +16,44 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import React, { useEffect, useState } from 'react';
+import React from 'react';
 import styled from 'styled-components';
 
-function SlideTabs({
+import { Indicator, Text } from '..';
+
+export function SlideTabs({
   appearance = 'square',
-  initialSelected = 0,
+  activeIndex = 0,
   name = 'slide-tab',
   onChange,
   size = 'xlarge',
   tabs,
+  isProcessing = false,
 }: props) {
-  const [activeIndex, setActiveIndex] = useState(initialSelected);
-
-  useEffect(() => {
-    onChange(activeIndex);
-  }, [activeIndex]);
-
   return (
     <Wrapper>
       <TabNav role="tablist" appearance={appearance} size={size}>
-        {tabs.map((tabData, tabIndex) => {
-          const tabDataType = typeof tabData === 'string';
-          const tabName = tabDataType ? tabData : tabData.name;
-          const tabContent = tabDataType ? tabData : tabData.component;
+        {tabs.map((tabName, tabIndex) => {
+          const selected = tabIndex === activeIndex;
           return (
             <TabLabel
               role="tab"
               htmlFor={`${name}-${tabName}`}
-              onClick={() => setActiveIndex(tabIndex)}
+              onClick={e => {
+                e.preventDefault();
+                onChange(tabIndex);
+              }}
               itemCount={tabs.length}
               key={`${tabName}-${tabIndex}`}
               className={tabIndex === activeIndex && 'selected'}
+              processing={isProcessing}
             >
-              {tabContent}
+              <Box>
+                {selected && isProcessing && (
+                  <Spinner delay="none" size="25px" />
```

Comment:
Because of the absolute positioning, the spinner will get rendered outside of the given tab when the browser width is narrow enough.

Though I understand that this will be used only in a single place, so I guess it's good enough for now.

I suppose the issue was that you didn't want the text of the tab to jump when the spinner gets shown? I wonder if a horizontal progress bar would work here. Connect has one in `LinearProgress.tsx`, but we don't have stories for it unfortunately. I see that there's a story for "Animated Progress Bar", but this might be too big idk.

Reply:
> I suppose the issue was that you didn't want the text of the tab to jump when the spinner gets shown?

yes

i'll add a note on this component to look into horizontal progress bar